### PR TITLE
Added configurable InputFields::ModernizedSelection::MaxNumberOfOptions.

### PR DESCRIPTION
**Pull request by itweserems**
## Proposed change
See #575:
When using dynamic fields of type "dropdown" or "multiselect", the default code-setup is that the layout of a select-field (e.g. dynamic field) will be changed when the amount of selectable values is above 1000.

This is for performance reasons, but the limit should be configurable. The layout for fields with >1000 values does not allow searching values and with a configurable maximum number can be set to an individual level to keep the search functionality as long as performance goes along with this.

## Type of change
1 - 🚀 feature

## Checklist

- [x] The code change is tested and works locally.(❗)
- [x] There is no commented out code in this PR.(❕)
- [ ] You improved or added new unit tests.(❕)
- [ ] Local ZnunyCodePolicy passed.(❕)
- [ ] Local UnitTests / Selenium passed.(❕)
- [ ] GitHub workflow CI (UnitTests / Selenium) passed.(❗)

### CODE REVIEW

**Comment by rkaldung** [comment]
@jepf Please check the new system configuration name, the feature looks fine.

File: Kernel/Output/HTML/Layout.pm [hunk 1]
```diff
@@ -508,6 +508,14 @@ sub Block {
...
         };
 
+    # For performance reasons:
+    # Do not initialize modern inputfields on selects with many entries
+    my $MaxDropdownOptions = $Kernel::OM->Get('Kernel::Config')->Get("MaxDropdownOptions");
+    $Self->AddJSData(
+        Key   => 'MaxDropdownOptions',
+        Value => $MaxDropdownOptions,
+    );
+
```
Comment (dennykorsukewitz) [inline Kernel/Output/HTML/Layout.pm]:
```suggestion
    # For performance reasons:
    # Do not initialize modernized input fields on selections with many entries
    my $MaxNumberOfOptions = $Kernel::OM->Get('Kernel::Config')->Get("InputFields::ModernizedSelection::MaxNumberOfOptions");
    $Self->AddJSData(
        Key   => 'InputFields::ModernizedSelection::MaxNumberOfOptions',
        Value => $MaxNumberOfOptions,
    );

```

File: Kernel/Language/de.pm [hunk 2]
```diff
@@ -5608,6 +5608,8 @@ sub Data {
             'Automatischer Zeilenumbruch in Textnachrichten nach x-Zeichen.',
         'Sets the number of lines that are displayed in text messages (e.g. ticket lines in the QueueZoom).' =>
             'Legt die Anzahl an Zeilen fest, die von Textnachrichten angezeigt werden sollen (z. B. von Tickets in der Queue-Ansicht).',
+        'Sets the maximum number of dropdown field values before the select layout will be changed for performance reasons.' =>
+            'Legt die maximale Anzahl an Werten fest, bevor das Auswahlfeld-Layout zu Performancezwecken geändert wird.',
```
Comment (dennykorsukewitz) [inline Kernel/Language/de.pm]:
```suggestion
        'Limits the number of options for modernized selection fields. If this limit will be exceeded, the selection field will for performance reasons be shown as non-modernized.' =>
            'Limitiert die Anzahl an Optionen für modernized Auswahlfelder. Wenn das Limit überschritten ist, wird das Auswahlfeld aus Performancegründen als nicht-modernized angezeigt.',
```

File: Kernel/Config/Files/XML/Framework.xml [hunk 3]
```diff
@@ -550,6 +550,13 @@
             <Item ValueType="String" ValueRegex="^[0-9]{1,5}$">6000</Item>
         </Value>
     </Setting>
+    <Setting Name="MaxDropdownOptions" Required="1" Valid="1">
+        <Description Translatable="1">Sets the maximum number of dropdown field values before the select layout will be changed for performance reasons.</Description>
+        <Navigation>Frontend::Base</Navigation>
+        <Value>
+            <Item ValueType="String" ValueRegex="^\d*$">1000</Item>
+        </Value>
+    </Setting>
```
Comment (dennykorsukewitz) [inline Kernel/Config/Files/XML/Framework.xml]:
```suggestion
    <Setting Name="InputFields::ModernizedSelection::MaxNumberOfOptions" Required="1" Valid="1">
        <Description Translatable="1">Limits the number of options for modernized selection fields. If this limit will be exceeded, the selection field will for performance reasons be shown as non-modernized.</Description>
        <Navigation>Frontend::Base</Navigation>
        <Value>
            <Item ValueType="String" ValueRegex="^\d+$">1000</Item>
        </Value>
    </Setting>
```

Please move this setting after:

```
    <Setting Name="ModernizeCustomerFormFields" Required="1" Valid="1">
        <Description Translatable="1">Use new type of select and autocomplete fields in customer interface, where applicable (InputFields).</Description>
        <Navigation>Frontend::Customer</Navigation>
        <Value>
            <Item ValueType="Checkbox">1</Item>
        </Value>
    </Setting>
```

File: var/httpd/htdocs/js/Core.UI.InputFields.js [hunk 4]
```diff
@@ -1189,7 +1188,8 @@ Core.UI.InputFields = (function (TargetNS) {
 
             // For performance reasons:
             // Do not initialize modern inputfields on selects with many entries
-            if ($(SelectObj).children('option').length > Config.MaxNumberOfOptions) {
+            MaxDropdownOptions = Core.Config.Get('MaxDropdownOptions');
+            if ($(SelectObj).children('option').length > MaxDropdownOptions) {
```
Comment (dennykorsukewitz) [inline var/httpd/htdocs/js/Core.UI.InputFields.js]:
```suggestion
            MaxNumberOfOptions = Core.Config.Get('InputFields::ModernizedSelection::MaxNumberOfOptions');
            if ($(SelectObj).children('option').length > MaxNumberOfOptions) {
```

**Review by dennykorsukewitz — CHANGES_REQUESTED** [review]
Hi  @itweserems ,
thanks for your PR.

I have a few change requests here.

Regards :rocket:

**Comment by dennykorsukewitz** [comment]
Hi @itweserems 

thank you for your PullRequest.
We will merge this PR and move this setting after following setting.

```
    <Setting Name="ModernizeCustomerFormFields" Required="1" Valid="1">
        <Description Translatable="1">Use new type of select and autocomplete fields in customer interface, where applicable (InputFields).</Description>
        <Navigation>Frontend::Customer</Navigation>
        <Value>
            <Item ValueType="Checkbox">1</Item>
        </Value>
    </Setting>
```

Good Job. 🚀

**Review by dennykorsukewitz — APPROVED** [review]

**Comment by itweserems** [comment]
Hi @dennykorsukewitz,
sorry for not responding on this but thanks for making the changes and merging this!

BR, Paul 🚀

**Comment by dennykorsukewitz** [comment]
> Hi @dennykorsukewitz, sorry for not responding on this but thanks for making the changes and merging this!
> 
> BR, Paul 🚀

Hi @itweserems 

I usually have more patience and wait, but I didn't want to leave this finished PullRequest lying around unnecessarily. 
Thanks for your contribution.

👍🏼 🚀

**Pull request merged**